### PR TITLE
Refactors Lich Phylactery to be a component, makes it actually work

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

- Refactors the Lich's Phylactery mechanic to be a component
  - The phylactery now works on signals instead of processing and waiting for their owner to die. 
- Fixes Lich Phylactery reviving not working
- Minor code improvements for things surrounding lich code
  - Made the IMBUE_SOUL signal return a bitflag instead of a boolean
- Also overall improves the UX for the phylactery. 
  - Better examine messages
  - Feedback for the lich when they die
- Updates the description for the bind soul spell to better reflect it's behavior. Also fixes said behavior. 
  - Apparently at some point someone completely reworked how Lich works but didn't update the spell description at all (FIVE YEARS AGO), leaving it to be completely misleading. I chose to restore part of the behavior and update the description to match how it should... probably work? I'm not entirely sure

## Why It's Good For The Game

Lich's phylactery straight up like, didn't work
This makes it work, and improves the code a good bit. 

## Changelog

:cl: Melbert
fix: Makes the wizard's Lich Phylactery mechanic actually work consistently. 
fix: The description of the Bind Soul spell (and entry) should match its behavior a bit more accurately now. You will revive regardless if your body is destroyed or leaves the z-level. With each revival, the time it takes to respawn does not change, but the duration of the stun does. 
refactor: Makes Phylacteries a component.
qol: Gave Lich Phylacerties some QoL features and pizazz. Examining it'll offer more information to wizards and ghosts.
/:cl:

### CODE REVIEW

File: code/datums/components/phylactery.dm [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+/datum/component/phylactery
+	// Set in initialize.
+	/// The mind of the lich who is linked to this phylactery.
+	var/datum/mind/lich_mind
+	/// The respawn timer of the phylactery.
+	var/base_respawn_time = 3 MINUTES
+	/// How much time is added on to the respawn time per revival.
+	var/time_per_resurrection = 1 MINUTES
+	/// The color of the phylactery itself. Applied on creation.
+	var/phylactery_color = "#003300"
```
Comment (SuperNovaa41) [inline code/datums/components/phylactery.dm:16]:
```suggestion
	var/phylactery_color = COLOR_VERY_DARK_LIME_GREEN
```

File: code/datums/components/phylactery.dm [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+	// Sweep up any revive signals left on the mind's current
+	UnregisterSignal(lich_mind.current, COMSIG_LIVING_REVIVE)
+
+	lich_mind = null
+	return ..()
+
+/**
+ * Signal proc for [COMSIG_PARENT_EXAMINE].
+ *
+ * Gives some flavor fort he phylactery on examine.
```
Comment (SuperNovaa41) [inline code/datums/components/phylactery.dm:72]:
```suggestion
 * Gives some flavor for the phylactery on examine.
```
wow

File: code/datums/components/phylactery.dm [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+/datum/component/phylactery/proc/stop_timer(mob/living/source, full_heal, admin_revive)
+	SIGNAL_HANDLER
+
+	deltimer(revive_timer)
+	revive_timer = null
+
+	UnregisterSignal(source, COMSIG_LIVING_REVIVE)
+
+/**
+ * Actuall undergo the process of reviving the lynch
```
Comment (SuperNovaa41) [inline code/datums/components/phylactery.dm:149]: The what now?

**Review by spookydonut — APPROVED** [review]

**Comment by timothymtorres** [comment]
@MrMelbert did this solve the issues for #62507, #37808, or #45518?

**Comment by MrMelbert** [comment]
fixed A, not sure about B, did not fix C

**Pull request merged**